### ISSUE TITLE
Smalltalk-80 on the ICL PERQ

### ISSUE THREAD

**Issue by pmcjones** [skeezicsb/PERQmedia#2]
I'm not sure if this would be of interest to you, but I ran across it today: an implementation of the Smalltalk-80 virtual machine, from the Blue Book, for the ICL Perq. The web site includes the author's master's thesis, plus his C source code, his development diary, the VM image from Xerox, and a transcription of the reference VM from the Blue Book.
    http://www.wolczko.com/st80/

Seen via: https://github.com/dbanay/Smalltalk (a modern C++ implementation of the same Blue Book VM) -- seen via: https://news.ycombinator.com/item?id=23307700

**Comment by skeezicsb** [comment]
Yes!  Thanks for the link.  I actually learned about his many years ago, but at the time Mario Wolczko thought the source code had been lost.  But just a few months ago Josh Dersch, author of PERQemu, found the web site and the source.  It's definitely on my list of PERQ projects, to see if it can be made to build and run again!

Are you the Paul McJones from the CHM?  I was looking at the blog yesterday and was going to write to you: there's another effort underway in the UK right now that may be of interest.  A version of the FLEX OS, written in ALGOL-68 and microcoded for the PERQ, is being recovered from floppy and hard disks from what may be the last surviving machine to run it.  It would be awfully cool to have _another_ unique OS environment (POS, MPOS, PNX, Accent and FLEX -- and we just found an HCR Unix port too?) available to explore for the PERQ.  It's great fun that after 40 years these things are still coming to light. :-)

**Comment by pmcjones** [comment]
Yes, I’m the Paul McJones who volunteers at CHM (e.g., http://www.softwarepreservation.org/projects and http://xeroxalto.computerhistory.org); also https://mcjones.org/dustydecks/ .

Could you say who you are? (I forgot this is going to be posted on github — email me privately at paul@mcjones.org if you are comfortable with that.)

Your new find sounds interesting indeed. Speaking of PERQ, is Brian Rosen still around? When I started at Xerox SDD (in 1976), Brian was there working on the Dolphin hardware. After a while he decided to return to Pittsburgh to work on the PERQ. I remember him calling me around 1981 asking if I’d like to join Three Rivers to work on PERQ software


Paul

**Comment by larsbrinkhoff** [comment]
CC @chrissamuel
